src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1485,6 +1485,38 @@ ss::future<> do_fetch(op_context& octx) {
 }
 } // namespace
 
+namespace testing {
+ss::future<> do_fetch(op_context& octx, model::fetch_read_strategy strategy) {
```

Comment:
instead of expanding the size of the redpanda executable, did you consider exposing the methods that this test code depends on?

Reply:
Good point, I've updated the testing method to simply forward to an existing internal method. This shouldn't increase the executable as much. I can also just directly expose the internal method outside of the `testing` namespace and avoid any increase. However, I'm not sure what our stance on that is from a code-cleanliness standpoint.